.github/workflows/_build-and-push.yml

```
@@ -0,0 +1,68 @@
+      - name: Extract metadata
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ${{ env.IMAGE_PREFIX }}/${{ matrix.image }}
+          tags: |
+            type=raw,value=main-${{ inputs.ref }}
+            type=raw,value=staging,enable={{is_default_branch}}
+            ${{ inputs.tags }}
+
+      - name: Build and Push ${{ matrix.image }} image
+        uses: docker/build-push-action@v4
```

actionlint (1.7.7)

the runner of "docker/build-push-action@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)